ext/openssl/openssl.c

```diff
@@ -4364,19 +4532,22 @@ PHP_FUNCTION(openssl_pkey_get_private)
 	EVP_PKEY *pkey;
 	char * passphrase = "";
 	size_t passphrase_len = sizeof("")-1;
-	zend_resource *res;
+	php_openssl_pkey_object *key_object;
 
 	if (zend_parse_parameters(ZEND_NUM_ARGS(), "z|s", &cert, &passphrase, &passphrase_len) == FAILURE) {
 		RETURN_THROWS();
 	}
 
 	PHP_OPENSSL_CHECK_SIZE_T_TO_INT(passphrase_len, passphrase);
-	pkey = php_openssl_evp_from_zval(cert, 0, passphrase, passphrase_len, 1, &res);
 
+	pkey = php_openssl_pkey_from_zval(cert, 0, passphrase, passphrase_len, NULL);
 	if (pkey == NULL) {
 		RETURN_FALSE;
 	}
-	ZVAL_RES(return_value, res);
+
+	object_init_ex(return_value, php_openssl_pkey_ce);
+	key_object = Z_OPENSSL_PKEY_P(return_value);
+	key_object->pkey = pkey;
```

Comment:
It might be good to create a macro for this as it's constantly repeated. Doesn't need to be in the PR - just a note for future improvement.